README.md

```diff
@@ -1,11 +1,203 @@
+    
+
+### `tiago_description`
+
+Contains the URDF, mesh, and configuration files for TIAGo.
+
+- **`config/`**: Contains configuration files like:
+    
+    - `tiago_configuration.yaml`: General configuration for the robot.
+        
+    - `tiago.rviz` & `urdf.rviz`: RViz visualization settings.
+        
+- **`launch/`**: Launch files for visualization and state publishing:
+    
+    - `robot_state_publisher.launch.py`
+        
+    - `show.launch.py`
+        
+- **`meshes/`**: 3D models of the robot components, including:
+    
+    - `arm/`, `head/`, `sensors/`, `torso/`
+        
+- **`module/`**: Contains `10_robot_state_publisher.yaml`, which defines robot state publishing settings.
+    
+- **`robots/`**: Contains `tiago.urdf.xacro`, the main Xacro definition of TIAGo.
+    
+- **`ros2_control/`**: ROS 2 control configurations, including:
+    
+    - `gazebo_controller_manager_cfg.yaml`
+        
+    - `ros2_control.urdf.xacro`
+        
+    - `transmissions.urdf.xacro`
+        
+- **`test/`**: Contains test scripts for validating TIAGo’s description:
+    
+    - `test_description.launch.py`
+        
+    - `test_xacro.py`
+        
+- **`tiago_description/`**: Contains Python utilities for launching TIAGo, including:
+    
+    - `launch_arguments.py`
+        
+    - `tiago_launch_utils.py`
+        
+- **`urdf/`**: Contains Xacro files defining specific robot components, structured as:
+    
+    - `arm/`, `head/`, `sensors/`, `torso/`, `end_effector/`, `calibration/`
+        
+    - Example: `torso/torso.ros2_control.xacro`, `torso/torso.urdf.xacro`
+        
+
+### `tiago_controller_configuration`
+
+Contains configuration files for controlling the robot’s actuators.
+
+- **`config/`**: YAML configurations for various controllers, including:
+    
+    - `arm_controller.yaml`: Controls the robot's arm.
+        
+    - `head_controller.yaml`: Controls the head movements.
+        
+    - `ft_sensor_controller.yaml`: Manages force-torque sensor data.
+        
+    - `joint_state_broadcaster.yaml`: Handles joint state broadcasting.
+        
+    - `gravity_compensation_controller_ilm.yaml`: Implements ILM-based gravity compensation.
+        
+    - `gravity_compensation_controller_parker.yaml`: Implements Parker-based gravity compensation.
+        
+    - `torso_controller.yaml`: Controls the torso movements.
+        
+- **`launch/`**: Launch files to initialize controllers, including:
+    
+    - `default_controllers.launch.py`: Loads and starts default controllers.
+        
+    - `gravity_compensation_controller.launch.py`: Launches gravity compensation controllers.
+        
+- **`module/`**: Contains YAML files specifying controller configurations:
+    
+    - `10_default_controllers.yaml`: Defines default controllers for initialization.
+        
+
+### `tiago_robot`
+
+Contains high-level package metadata, including `package.xml` and `CMakeLists.txt`.
```

Comment:
I don't think you need to explain each and every file, just give an overview of what each package does